src/probnum/diffeq/perturbedsolvers/_perturbation_functions.py

```diff
@@ -0,0 +1,50 @@
+"""perturbation functions to perturb the stepsize."""
+import numpy as np
+
+
+def perturb_uniform(step, order, noise_scale, seed=1):
+    """perturbs the step with uniformly distributed noise scaled by noise-scale.
+    proposed by Abdulle and Garegnani(2020)
```

Comment:
docstrings are usually written in imperative: "Perturb the step with uniformly...". 

Regarding the reference: there is an official sphinx environment thingy for "Reference", which may be useful here? The docstrings for "probsolve_ivp" or "problinsolve" would be a good starting point to learn how (well, at least this is where I look it up because I forget it all the time)